Dockerfile

```diff
@@ -1,5 +1,12 @@
+FROM golang:1.10 as builder
+WORKDIR /go/src/github.com/containous/whoami
+COPY . .
+RUN go get -u github.com/golang/dep/cmd/dep
+RUN make dependencies
+RUN make build
+
 # Create a minimal container to run a Golang static binary
 FROM scratch
-COPY whoami /
+COPY --from=builder /go/src/github.com/containous/whoami/whoami .
 ENTRYPOINT ["/whoami"]
-EXPOSE 80
+EXPOSE 80
```

Comment:
Sorry to nitpick here but as http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_206 , a new line is required at the end of the file.